lib/docker.js

```diff
@@ -130,14 +130,26 @@ function createDockerFile() {
     }
 
     contents = 'FROM ' + baseImg + '\n' +
-        'RUN apt-get update && apt-get install -y ' + extraPkgs.join(' ') +
+        'RUN apt-get update --fix-missing && apt-get install -y ' + extraPkgs.join(' ') +
         ' && rm -rf /var/lib/apt/lists/*\n';
 
+    var nodeVersion = pkg.deploy.node;
+    if (nodeVersion && nodeVersion !== 'system') {
+        var nvmDownloadURI = 'https://raw.githubusercontent.com/creationix/nvm/v0.29.0/install.sh';
+        contents += 'ENV NVM_DIR /usr/local/nvm\n';
+        contents += 'RUN wget -qO- ' + nvmDownloadURI + ' | bash' +
+                    ' && . $NVM_DIR/nvm.sh' +
+                    ' && nvm install ' + nodeVersion +
+                    ' && nvm use ' + nodeVersion + '\n';
+        contents += 'ENV NODE_PATH=$NVM_DIR/v' + nodeVersion + '/lib/node_modules ' +
+                    'PATH=$NVM_DIR/v' + nodeVersion + '/bin:$PATH\n';
+    }
+
     if (!opts.deploy) {
         contents += 'RUN mkdir /opt/service\n' +
             'ADD . /opt/service\n' +
             'WORKDIR /opt/service\n' +
-            'RUN npm install\n';
+            'RUN rm -rf node_modules; npm install\n';
```

Comment:
That should be specified in a `.dockerignore` file. All of the services based off of the template [have it](https://github.com/wikimedia/service-template-node/blob/master/.dockerignore), we should add it to RESTBase as well and leave this out.